### PR TITLE
fix image.Match to take only repo or repo:tag

### LINKED ISSUE #740: Bug: Image.Match() matching image using prefix influence some api
### BUG reproduce
one manage: 127.0.0.1:2222
one node with busybox image:127.0.0.1:2375
* docker rmi API
```
[root@localhost swarm]# docker -H 127.0.0.1:2222 rmi bus
Error response from daemon: localhost.localdomain: 409 Conflict: Conflict, cannot delete image 4986bf8c15363d1c5d15512d5266f8777bfba4974ac56e3270e7760f6f0a8125 because it is tagged in multiple repositories, use -f to force
FATA[0000] Error: failed to remove one or more images 

[root@localhost swarm]# docker -H 127.0.0.1:2222 rmi not_exists
Error response from daemon: No such image not_exists
FATA[0000] Error: failed to remove one or more images

[root@localhost swarm]# docker -H 127.0.0.1:2375 rmi bus
Error response from daemon: No such image: bus
FATA[0000] Error: failed to remove one or more images  
```
We find that "bus image" and "not_exists image"  are not in node, but respond differently. Because the busybox'prefix is bus, in our swarm, will send the request the node with busybox.

* docker history API
```
[root@localhost swarm]# docker -H 127.0.0.1:2222 history bus
FATA[0000] Error response from daemon: could not find image: no such id: bus 
[root@localhost swarm]# docker -H 127.0.0.1:2222 history not_exist
FATA[0000] Error response from daemon: No such image: not_exist 
[root@localhost swarm]# docker -H 127.0.0.1:2375 history bus
FATA[0000] Error response from daemon: could not find image: no such id: bus
```
Also respond differently

* docker save API
```
[root@localhost swarm]# docker -H 127.0.0.1:2222 save bus > /home/aaa.tar
time="2015-05-07T06:14:01-04:00" level="fatal" msg="Error: could not find image: no such id: bus" 
[root@localhost swarm]# 
[root@localhost swarm]# docker -H 127.0.0.1:2222 save not_exists > /home/aaa.tar
time="2015-05-07T06:14:11-04:00" level="fatal" msg="Error: No such image: not_exists" 
```
Also respond differently

* Other API which use proxyImage and Image.Match() maybe have the save problem

### Problem
Image.Match() : match image name using prefix

### How to Solve? Need help
Fix matching image name behavior. 
Sometimes name contains tag, sometimes do not have. How to match, does not have a clear way?
In delete image, if name do not have tag, docker will delete latest.
In save image, if name do not have tag, will save all images with repoName. if have tag, will save that one.

### PR DESCRIPTION
**Pull request by vieux**
close #740

### CODE REVIEW

File: test/integration/api/rmi.bats [hunk 1]
```diff
@@ -47,3 +47,12 @@ function teardown() {
...
 }
+
+@test "docker rmi prefix" {
+	start_docker_with_busybox 1
+	swarm_manage
+
+	run docker_swarm rmi bus
+	[ "$status" -ne 0 ]
+	[[ "${output}" == *"No such image"* ]]
+}
```
Comment (aluzzardi) [inline test/integration/api/rmi.bats]: What is this weird looking character in GH? :)

**Comment by aluzzardi** [comment]
What about `docker history busy`? Does docker support that? I personally don't care much it seems silly, but anyway

File: test/integration/api/rmi.bats (hunk 1 above)
Comment (vieux) [inline test/integration/api/rmi.bats]: no new line at the end I'll fix

**Comment by vieux** [comment]
@aluzzardi it it doesn't docker only supports looking by "prefix" for IDs, never names.

It's mixed in match already, not sure it's worth to add a new test.

File: test/integration/api/rmi.bats (hunk 1 above)
Comment (vieux) [inline test/integration/api/rmi.bats]: your magic "shell lint" should check that :D

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**